### ISSUE TITLE
Ajax

### ISSUE THREAD

**Issue by bhagwatsolunke** [bhagwatsolunke/lab#2]
<!DOCTYPE html>
<html>
<head>
  <title>User Registration</title>
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
  <script src="script.js"></script>
</head>
<body>
  <h1>User Registration</h1>
  <form id="registrationForm">
    <label for="name">Name:</label>
    <input type="text" id="name" required><br><br>
    <label for="email">Email:</label>
    <input type="email" id="email" required><br><br>
    <button type="submit">Register</button>
  </form>
</body>
</html>

$(document).ready(function() {
  // Check if local storage is supported
  var isLocalStorageSupported = function() {
    try {
      var testKey = 'test';
      localStorage.setItem(testKey, testKey);
      localStorage.removeItem(testKey);
      return true;
    } catch (e) {
      return false;
    }
  };

  // Function to store data in local storage
  var storeDataInLocalStorage = function(data) {
    if (isLocalStorageSupported()) {
      var storedData = localStorage.getItem('registeredUsers');
      if (storedData) {
        storedData = JSON.parse(storedData);
      } else {
        storedData = [];
      }
      storedData.push(data);
      localStorage.setItem('registeredUsers', JSON.stringify(storedData));
    } else {
      console.log('Local storage is not supported in this browser.');
    }
  };

  // Function to store data in an array
  var storeDataInArray = function(data) {
    var registeredUsers = [];
    if (Array.isArray(data)) {
      registeredUsers = data;
    }
    registeredUsers.push(data);
  };

  // Function to handle form submission
  var handleFormSubmission = function(event) {
    event.preventDefault();

    var name = $('#name').val();
    var email = $('#email').val();

    var formData = {
      name: name,
      email: email
    };

    // Store data in local storage
    storeDataInLocalStorage(formData);

    // Alternatively, store data in an array
    // storeDataInArray(formData);

    // Send data to server using AJAX POST
    $.ajax({
      url: 'your-server-endpoint', // Replace with your server endpoint URL
      type: 'POST',
      data: formData,
      success: function(response) {
        console.log('Data sent to the server successfully!');
        // Redirect to the new page that displays the list of registered users
        window.location.href = 'users.html'; // Replace with the URL of the page you want to redirect to
      },
      error: function(error) {
        console.log('An error occurred while sending data to the server.');
      }
    });
  };

  // Bind the form submission event to the handler function
  $('#registrationForm').submit(handleFormSubmission);
});

**Comment by bhagwatsolunke** [comment]
Sure! Here's an example of a JavaScript program that allows the user to input their registration data, sends it to the server using AJAX POST method, and stores the data in either an array or the browser's local storage. It also redirects the user to a new page that displays the list of registered users.

First, let's assume you have an HTML file with the following structure:

```html
<!DOCTYPE html>
<html>
<head>
  <title>User Registration</title>
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
  <script src="script.js"></script>
</head>
<body>
  <h1>User Registration</h1>
  <form id="registrationForm">
    <label for="name">Name:</label>
    <input type="text" id="name" required><br><br>
    <label for="email">Email:</label>
    <input type="email" id="email" required><br><br>
    <button type="submit">Register</button>
  </form>
</body>
</html>
```

Create a JavaScript file named "script.js" with the following code:

```javascript
$(document).ready(function() {
  // Check if local storage is supported
  var isLocalStorageSupported = function() {
    try {
      var testKey = 'test';
      localStorage.setItem(testKey, testKey);
      localStorage.removeItem(testKey);
      return true;
    } catch (e) {
      return false;
    }
  };

  // Function to store data in local storage
  var storeDataInLocalStorage = function(data) {
    if (isLocalStorageSupported()) {
      var storedData = localStorage.getItem('registeredUsers');
      if (storedData) {
        storedData = JSON.parse(storedData);
      } else {
        storedData = [];
      }
      storedData.push(data);
      localStorage.setItem('registeredUsers', JSON.stringify(storedData));
    } else {
      console.log('Local storage is not supported in this browser.');
    }
  };

  // Function to store data in an array
  var storeDataInArray = function(data) {
    var registeredUsers = [];
    if (Array.isArray(data)) {
      registeredUsers = data;
    }
    registeredUsers.push(data);
  };

  // Function to handle form submission
  var handleFormSubmission = function(event) {
    event.preventDefault();

    var name = $('#name').val();
    var email = $('#email').val();

    var formData = {
      name: name,
      email: email
    };

    // Store data in local storage
    storeDataInLocalStorage(formData);

    // Alternatively, store data in an array
    // storeDataInArray(formData);

    // Send data to server using AJAX POST
    $.ajax({
      url: 'your-server-endpoint', // Replace with your server endpoint URL
      type: 'POST',
      data: formData,
      success: function(response) {
        console.log('Data sent to the server successfully!');
        // Redirect to the new page that displays the list of registered users
        window.location.href = 'users.html'; // Replace with the URL of the page you want to redirect to
      },
      error: function(error) {
        console.log('An error occurred while sending data to the server.');
      }
    });
  };

  // Bind the form submission event to the handler function
  $('#registrationForm').submit(handleFormSubmission);
});
```

In this code, replace `'your-server-endpoint'` in the AJAX call with the actual endpoint URL where you want to send the data.

Now, you need to create a new HTML file named "users.html" that will
